chia/util/api_decorators.py

```diff
@@ -34,104 +26,66 @@ class ApiMetadata:
     peer_required: bool = False
     bytes_required: bool = False
     execute_task: bool = False
-    reply_type: List[ProtocolMessageTypes] = field(default_factory=list)
+    reply_types: List[ProtocolMessageTypes] = field(default_factory=list)
     message_class: Optional[Any] = None
 
 
-def get_metadata(function: Callable[..., Any]) -> ApiMetadata:
+def get_metadata(function: Callable[..., object]) -> ApiMetadata:
     maybe_metadata: Optional[ApiMetadata] = getattr(function, metadata_attribute_name, None)
     if maybe_metadata is None:
         return ApiMetadata()
 
     return maybe_metadata
```

Comment:
```suggestion
def get_metadata(function: Callable[..., object]) -> Optional[ApiMetadata]:
    return getattr(function, metadata_attribute_name, None)
```

I think ^ to return an `Optional[ApiMetadata]` and then drop `ApiMetadata.api_function` member is cleaner so that we can just check for methods which have an actual metadata assigned instead of returning this default object. Thoughts?

Reply:
There is simplicity in both options.  I haven't managed to come up with a clear winner.  I'll switch it.